localstack-core/localstack/services/events/target.py

```diff
@@ -181,25 +181,20 @@ def _validate_input(self, target: Target):
             self._validate_input_transformer(input_transformer)
 
     def _initialize_client(self) -> BaseClient:
-        """Initializes internal boto client.
-        If a role from a target is provided, the client will be initialized with the assumed role.
-        If no role is provided, the client will be initialized with the account ID and region.
-        In both cases event bridge is requested as service principal"""
+        """Initializes internal boto client."""
         service_principal = ServicePrincipal.events
         role_arn = self.target.get("RoleArn")
-        if role_arn:  # required for cross account
-            # assumed role sessions expire after 6 hours in AWS, currently no expiration in LocalStack
+        if role_arn:
             client_factory = connect_to.with_assumed_role(
                 role_arn=role_arn,
                 service_principal=service_principal,
                 region_name=self.region,
             )
+            client = client_factory.get_client(self.service)
```

Comment:
what is the reason behind reversing the behavior - the idea is that if a role_arn is provided it is used as the source arn and not the rule_arn as initially implemented.

Reply:
hey, thanks for catching these. I had a issue related to JAVA_PATH and to solve it I tried few things. I committed these changes accidentally. I have reverted them now.

Reply:
yes JAVA_PATH running it locally on mac is a problem
did you fix it?
I always hardcode my java home and change the ending to dylib

```
 java_home = "/usr/local/Cellar/openjdk/23/libexec/openjdk.jdk/Contents/Home"
    jvm_lib = Path(java_home) / "lib" / "server" / "libjvm.dylib"
```